src/renderer/components/dock/terminal.store.ts

```diff
@@ -104,18 +105,42 @@ export class TerminalStore extends Singleton {
     return this.connections.get(tabId)?.readyState === WebSocketApiState.CLOSED;
   }
 
-  sendCommand(command: string, options: { enter?: boolean; newTab?: boolean; tabId?: TabId } = {}) {
+  async sendCommand(command: string, options: { enter?: boolean; newTab?: boolean; tabId?: TabId } = {}) {
     const { enter, newTab, tabId } = options;
-    const { selectTab, getTabById } = dockStore;
-    const tab = tabId && getTabById(tabId);
 
-    if (tab) selectTab(tabId);
-    if (newTab) createTerminalTab();
+    if (tabId) {
+      dockStore.selectTab(tabId);
+    }
+
+    if (newTab) {
+      const tab = createTerminalTab();
+
+      await when(() => this.connections.has(tab.id));
+
+      const rcIsFinished = when(() => this.connections.get(tab.id).rcFinished);
+      const notifyLong = setTimeout(() => {
+        Notifications.info("Terminal shell is taking a long time to complete startup. Please check your .rc file.", {
+          timeout: 4_000,
+        });
+      }, 2_500);
+      const notifyVeryLong = setTimeout(() => {
+        rcIsFinished.cancel();
+        Notifications.info("Bypassing shell completion check.", {
+          timeout: 4_000,
+        });
+      }, 7_500);
```

Comment:
Seems like between 2 timeouts just 5 seconds of difference. Maybe worth combine those 2 notifications into 1?

Reply:
Sounds good.